main/src/ca/uwaterloo/flix/language/phase/ClosureConv.scala

```diff
@@ -308,6 +381,14 @@ object ClosureConv {
         val es = exps.map(visitExp)
         Expr.ApplyDef(sym, es, tpe, purity, loc)
 
+      case Expr.ApplyLocalDef(sym, exps, itpe, tpe, purity, loc) =>
+        val es = exps.map(visitExp)
+        val sym1 = subst.get(sym) match {
```

Comment:
I think this implies that 
```
def f(x) = x;
def g(x) = f(x);
g(12)
```
will become
```
def f(x) = x;
def g(q, x) = q(x);
g(f, x)
```
Or something like that? Wouldn't that mean that the `ApplyLocalDef` now contains a normal varsym and that (less importantly) we don't have a direct call?

Reply:
Why are binders are replaced by subst at all? seems weird

Reply:
Good point

Reply:
But then local def symbols can never be free, i.e., in `freevars` we should always exclude the symbol of a local def... but then a lambda may capture it? I guess that's okay since it's replaced by a `DefnSym` in `LambdaLift`